source/ingress.go

```diff
@@ -152,7 +160,11 @@ func endpointsFromIngress(ing *v1beta1.Ingress) []*endpoint.Endpoint {
 			continue
 		}
 
-		endpoints = addEndpointsFromTargetAnnotation(ing, rule.Host, endpoints)
+		endpoints = getEndpointsFromTargetAnnotation(ing, rule.Host)
```

Comment:
IMHO `getEndpointsFromTargetAnnotation` should be called from `Endpoint()` method above: 
1. Try to generate endpoint from target annotation key, if none: 
2. Try to generate endpoint from ingress status field, if none:
3. Try to generate endpoint from template